src/backend/distributed/commands/owned.c

```diff
@@ -89,3 +92,74 @@ PreprocessDropOwnedStmt(Node *node, const char *queryString,
 
 	return NodeDDLTaskList(NON_COORDINATOR_NODES, commands);
 }
+
+
+/*
+ * Post-processes a REASSIGN OWNED statement.
+ *
+ * This function takes a Node pointer representing a REASSIGN OWNED statement,
+ * and performs any necessary post-processing after the statement has been executed.
+ * In this method, we propagate all the roles that are being reassigned.
+ * Therefore, we don't filter the roles based on whether they are distributed or not.
+ * If roles are not distributed and those roles are not present in the nodes
+ * where the statement is being propagated, then the statement will fail.
+ *
+ * Parameters:
+ *   stmt: A pointer to a Node representing a REASSIGN OWNED statement.
+ *   queryString: The original SQL command string from the client.
+ *
+ * Returns:
+ *   List of SQL statements to be executed.
+ */
+List *
+PostprocessReassignOwnedStmt(Node *node, const char *queryString)
+{
+	ReassignOwnedStmt *stmt = castNode(ReassignOwnedStmt, node);
+	List *allReassignRoles = stmt->roles;
+
+	List *distributedReassignRoles = FilterDistributedRoles(allReassignRoles);
+
+	if (list_length(distributedReassignRoles) <= 0)
+	{
+		return NIL;
+	}
+
+	if (!ShouldPropagate())
+	{
+		return NIL;
+	}
+
+	EnsureCoordinator();
+
+	stmt->roles = distributedReassignRoles;
+	char *sql = DeparseTreeNode((Node *) stmt);
+	stmt->roles = allReassignRoles;
+
+	ObjectAddress *newRoleAddress = GetNewRoleAddress(stmt);
+
+	int saveNestLevel = NewGUCNestLevel();
+	set_config_option("citus.enable_create_role_propagation", "on",
+					  (superuser() ? PGC_SUSET : PGC_USERSET), PGC_S_SESSION,
+					  GUC_ACTION_LOCAL, true, 0, false);
```

Comment:
We never do anything when calling EnsureDependencies function, and sometimes we may need to propagate roles in that function. Whatever we did before, the same can be applied here. So, I think we don't need these lines.

Reply:
This block is required to set the GUC into previous state. I've used it in citus_internal_database_command
https://github.com/citusdata/citus/blob/18a8783e96983aef1c780d8019d5ce148b3dcad2/src/backend/distributed/metadata/metadata_sync.c#L4009